tendrl/commons/objects/node/atoms/valid_node_details/__init__.py

```diff
@@ -0,0 +1,61 @@
+import etcd
+
+from tendrl.commons import objects
+from tendrl.commons.objects import AtomExecutionFailedError
+from tendrl.commons.objects.node import Node
+
+
+class ValidNodeDetails(objects.BaseAtom):
+    obj = Node
+
+    def __init__(self, *args, **kwargs):
+        super(ValidNodeDetails, self).__init__(*args, **kwargs)
+
+    def run(self):
+        import pdb; pdb.set_trace()
+        node_ids = self.parameters.get('Node[]')
+        if not node_ids or len(node_ids) == 0:
+            raise AtomExecutionFailedError("Node[] cannot be empty")
+
+        for node_id in node_ids:
+            # Check if node has the OS details populated
+            try:
+                NS._int.client.read("nodes/%s/Os" % node_id)
+            except etcd.EtcdKeyNotFound:
+                raise AtomExecutionFailedError(
+                    "Node doesnt have OS details populated"
+                )
+
+            # Check if node has the CPU details populated
+            try:
+                NS._int.client.read("nodes/%s/Cpu" % node_id)
+            except etcd.EtcdKeyNotFound:
+                raise AtomExecutionFailedError(
+                    "Node doesnt have CPU details populated"
+                )
+
+            # Check if node has the Memory populated
+            try:
+                NS._int.client.read("nodes/%s/Memory" % node_id)
+            except etcd.EtcdKeyNotFound:
+                raise AtomExecutionFailedError(
+                    "Node doesnt have Memory details populated"
+                )
+
+            # Check if node has the block devices populated
+            try:
+                NS._int.client.read("nodes/%s/BlockDevices" % node_id)
```

Comment:
You'll have to check /Disks too.

Also, youll need to add checks if these /Disks or any other directories exist and are **not empty**  (meaning result.leaves is not None)